charms/argo-controller/terraform/main.tf

```diff
@@ -0,0 +1,13 @@
+resource "juju_application" "argo_controller" {
+  charm {
+    name     = "argo-controller"
+    channel  = var.channel
+    revision = var.revision
+  }
+  config    = var.config
+  model     = var.model_name
+  name      = var.app_name
+  resources = var.resources
+  trust     = true
+  units     = 1
```

Comment:
Feels like the scale of the deployed charm should be an input variable for the module

We use 'scale' in Sunbeam as an example:

https://github.com/canonical/sunbeam-terraform/blob/main/modules/openstack-api/variables.tf#L39

but you might also want to model this as a top-level boolean type variable to enable high-availability - i.e units == 1 or units == 3 if that makes sense.

Reply:
@javacruft thanks for taking a look at the PR. What you describe makes sense but the `units` input variable is omitted on purpose because we haven't tested and confirmed that our charms work with more than one unit. I have this note in the PR's description:

>Omitted units from variables (inputs). This is because CKF charms have not been designed to function with more than one units.